### PR TITLE
Added ability for anchor menu items to have sub menus.

### LINKED ISSUE #4196: Anchor Menu Items cannot contain sub menus
### Bug Report

- __Package version(s)__: latest

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? Yes

Requested priority: Normal

Products/sites affected: VSTS

#### Describe the issue:

Currently there is no ability to have a menu item that is both an anchor element, and contains a sub menu. The current workaround is to use a button to get the sub menu, but in that case we lose the ability to right-click, open in a new tab, etc.

#### Actual behavior:

When a menu item contains an href and subMenuProps, the subMenuProps are ignored.

#### Expected behavior:

When a menu item contains an href and subMenuProps, an <a> tag is used and the sub menu is rendered.

### PR DESCRIPTION
**Pull request by dmarcey**
Added prop for subMenuHoverDelay.

#### Pull request checklist

- [x] Addresses an existing issue: Fixes #4196, #4197
- [x] Include a change request file using `$ npm run change`

#### Description of changes

- Allow anchor menu items to have sub menus. 
- Added a prop for sub menu hover delay.
- Added examples for each.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx [hunk 1]
```diff
@@ -784,6 +799,7 @@ export class ContextualMenu extends BaseComponent<IContextualMenuProps, IContext
    */
   private _updateFocusOnMouseEvent(item: any, ev: React.MouseEvent<HTMLElement>) {
     const targetElement = ev.currentTarget as HTMLElement;
+    const timeoutDuration = this.props.subMenuHoverDelay !== undefined ? this.props.subMenuHoverDelay : this._navigationIdleDelay;
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: `const { subMenuHoverDelay: timeoutDuration = this._navigationIdleDelay } = this.props;`?

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.types.ts [hunk 2]
```diff
@@ -137,7 +137,7 @@ export interface IContextualMenuProps extends React.Props<ContextualMenu>, IWith
    * Click handler which is invoked if onClick is not passed for individual contextual
    * menu item
    */
-  onItemClick?: (ev?: React.MouseEvent<HTMLElement>, item?: IContextualMenuItem) => void;
+  onItemClick?: (ev?: React.MouseEvent<HTMLElement>, item?: IContextualMenuItem) => boolean | void;
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.types.ts]: add comment explaining the return type?

**Comment by joschect** [comment]
@dmarcey Let's move this logic into split button. Really only split buttons should have submenu and a button.

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx [hunk 3]
```diff
@@ -470,7 +470,16 @@ export class ContextualMenu extends BaseComponent<IContextualMenuProps, IContext
   }
 
   private _renderAnchorMenuItem(item: IContextualMenuItem, classNames: IMenuItemClassNames, index: number, focusableElementIndex: number, totalItemCount: number, hasCheckmarks: boolean, hasIcons: boolean): React.ReactNode {
+    const { expandedMenuItemKey } = this.state;
     const { contextualMenuItemAs: ChildrenRenderer = ContextualMenuItem } = this.props;
+
+    let { subMenuId } = this.state;
+    if (item.subMenuProps && item.subMenuProps.id) {
```
Comment (joschect) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: Maybe put this in a method since this logic is used elsewhere.

Comment (dmarcey) [reply packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: Fixed. Thanks.

**Pull request merged**